libs/java/service/src/main/java/ca/bc/gov/tno/services/data/TnoApi.java

```diff
@@ -0,0 +1,218 @@
+    this.rest = new RestTemplate();
+
+    var pmap = new HashMap<String, String>(4);
+    pmap.put("charset", "utf-8");
+    this.jsonMediaType = new MediaType("application", "json", pmap);
+
+    var xml = new MappingJackson2HttpMessageConverter();
+    xml.setSupportedMediaTypes(Arrays.asList(new MediaType[] { MediaType.TEXT_XML }));
+    var json = new MappingJackson2HttpMessageConverter();
+    json.setSupportedMediaTypes(
+        Arrays.asList(
+            new MediaType[] { MediaType.APPLICATION_JSON, jsonMediaType }));
+    var form = new FormHttpMessageConverter();
+    rest.setMessageConverters(Arrays.asList(new HttpMessageConverter<?>[] { json, xml, form }));
+  }
+
+  /**
+   * @return The current access token.
+   */
+  public TokenResponse getToken() {
+    return this.token;
+  }
+
+  /**
+   * Make AJAX request to keycloak to fetch access token.
+   * 
+   * @return The token response.
+   * @throws ApiException If the AJAX request fails for some reason.
+   */
+  public TokenResponse requestToken() throws ApiException {
+    var url = config.getAuthorityUrl() + "/realms/" + config.getRealm() + "/protocol/openid-connect/token";
+    var headers = new HttpHeaders();
+    headers.setAccept(Arrays.asList(MediaType.APPLICATION_JSON));
+    headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
+    var map = new LinkedMultiValueMap<String, String>();
+    map.add("audience", config.getClientId());
+    map.add("grant_type", "client_credentials");
+    map.add("client_id", config.getClientId());
+    map.add("client_secret", config.getClientSecret());
+    var entity = new HttpEntity<MultiValueMap<String, String>>(map, headers);
+    var response = rest.exchange(url, HttpMethod.POST, entity, TokenResponse.class);
+
+    var status = response.getStatusCode();
+    if (status.series() == HttpStatus.Series.SUCCESSFUL) {
+      this.token = response.getBody();
+      return this.token;
+    } else {
+      throw new ApiException(String.format("Failed to retrieve token: %s", status.getReasonPhrase()));
+    }
+  }
+
+  /**
+   * Make an AJAX request to the API.
+   * 
+   * @param <T>          The type of object.
+   * @param url          The URL to the endpoint.
+   * @param method       The HTTP method.
+   * @param data         The request body data.
+   * @param responseType The response type.
+   * @return A new instance of an object of the specified type.
+   * @throws ApiException            If an error occurred when communicating with
+   *                                 the API.
+   * @throws ResourceAccessException An error occurred while making an AJAX
+   *                                 request.
+   */
+  public <T> T request(String url, HttpMethod method, T data, Class<T> responseType)
+      throws ApiException, ResourceAccessException {
+    var token = requestToken();
+
+    var headers = new HttpHeaders();
+    headers.setAccept(Arrays.asList(MediaType.APPLICATION_JSON, jsonMediaType));
```

Comment:
I have explicitly told it to accept `application/json?charset=utf-8` but it doesn't work...